proc-macros/src/render_server.rs

```diff
@@ -107,6 +107,22 @@ impl RpcDescription {
 		})
 	}
 
+	/// Helper that will ignore results of `register_*` method calls, and panic if there have been
+	/// any errors in debug builds.
+	///
+	/// The debug assert is a safeguard should the contract that guarantees the method names to
+	/// never conflict in the macro be broken in the future.
+	fn handle_register_result(&self, tokens: TokenStream2) -> TokenStream2 {
+		let reexports = self.jrps_server_item(quote! { core::__reexports });
+		quote! {{
+			let _res = #tokens;
+			#[cfg(debug_assertions)]
+			if _res.is_err() {
+				#reexports::panic_fail_register();
+			}
```

Comment:
What is the advantage of moving to `#[cfg(debug_assertions)]` with a separate panic over `debug_assert!`? It looks like they should both be equivalent to me.

Also is there any reason to move from `res` to `_res`?

Reply:
Less macro expansion at compile time and less code at callsites, see previous comments

Reply:
Ah I did the cargo expand myself; I was expecting both macros to elide the code in release mode but indeed only the `#[cfg(debug_assertions)]` seems too!